bimsb/packages/tainted.scm

```diff
@@ -196,16 +198,18 @@ deviation (SD) plots, coefficient of variation (CV) plots.")
                   "src/quantifier.pl"))
                ;; patch script using ../Rfam_for_miRDeep.fa file
                (substitute* "src/miRDeep2.pl"
-                 (("\\$\\{scripts\\}/\\.\\./Rfam_for_miRDeep\\.fa")
-                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))
-                 (("\\$scripts/\\.\\./Rfam_for_miRDeep\\.fa")
-                  (string-append out "/share/mirdeep2/Rfam_for_miRDeep.fa"))))
+                 (("\\$\\{?scripts\\}?/\\.\\.(/Rfam_for_miRDeep\\.fa)" full file)
+                  (string-append out "/share/mirdeep2" file))
+                 ;; Comment out all lines featuring the `scripts` variable for
+                 ;; paths relative to the script.
+                 ((".*\\$scripts.*" line)
+                  (string-append "#" line))))
              ;; patch script using $(dirname `which miRDeep2.pl`)/indexes dir
              (substitute* "src/make_html.pl"
                (("`which miRDeep2\\.pl`")
                 "`realpath ~/.local/share/mirdeep2` . \"/\"")
-               (("mkdir \"\\$\\{scripts\\}indexes")
-                "mkdir -p \"${scripts}indexes"))
+               (("(mkdir) (\"\\$\\{scripts\\}indexes)" call cmd path)
+                (string-append cmd " -p " path)))
```

Comment:
My ever-growing gut tells me that repeating "mkdir" is preferable to complicating the regular expression with more matching.

There is an alternative to all of this: we could write a proper patch, drop it in the root directory of the repository, and apply it in the `source` field.  We could even come up with a patch that upstream would accept, e.g. running `which` only once and using the result everywhere.  Then the amount of patching we have to maintain would become much smaller.

Reply:
Sure, I can replace `cdm "` by  `"mkdir` and drop the extra parents in the pattern. ;-)

I can also write a patch. Just didn't learn how to apply with Guix yet. Do you have an example for how to do that via `source` at hand? Otherwise I'll check the manual myself.  But would prefer that over all the substitutions via scheme. I doubt upstream would expect it though. But feel free to contact @drmirdeep once we have it working here.

Reply:
I went with the first solution for now as I don't know when I'll find time to work on a proper patch.
Any improvement is better than none, I guess. ;-)